packages/react-reconciler/src/__tests__/useMutableSource-test.internal.js

```diff
@@ -0,0 +1,1250 @@
+              label="b"
+              getSnapshot={defaultGetSnapshot}
+              mutableSource={mutableSource}
+              subscribe={defaultSubscribe}
+            />
+          </>,
+          () => Scheduler.unstable_yieldValue('Sync effect'),
+        );
+        expect(Scheduler).toFlushAndYield(['a:one', 'b:one', 'Sync effect']);
+      });
+    });
+
+    it('should throw and restart render if source and snapshot are unavailable during an update', () => {
+      const source = createSource('one');
+      const mutableSource = createMutableSource(source);
+
+      act(() => {
+        ReactNoop.render(
+          <>
+            <Component
+              label="a"
+              getSnapshot={defaultGetSnapshot}
+              mutableSource={mutableSource}
+              subscribe={defaultSubscribe}
+            />
+            <Component
+              label="b"
+              getSnapshot={defaultGetSnapshot}
+              mutableSource={mutableSource}
+              subscribe={defaultSubscribe}
+            />
+          </>,
+          () => Scheduler.unstable_yieldValue('Sync effect'),
+        );
+        expect(Scheduler).toFlushAndYield(['a:one', 'b:one', 'Sync effect']);
+        ReactNoop.flushPassiveEffects();
+
+        // Changing values should schedule an update with React.
+        // Start working on this update but don't finish it.
+        Scheduler.unstable_runWithPriority(
+          Scheduler.unstable_LowPriority,
+          () => {
+            source.value = 'two';
+            expect(Scheduler).toFlushAndYieldThrough(['a:two']);
+          },
+        );
+
+        const newGetSnapshot = s => 'new:' + defaultGetSnapshot(s);
+
+        // Force a higher priority render with a new config.
+        // This should signal that the snapshot is not safe and trigger a full re-render.
+        Scheduler.unstable_runWithPriority(
+          Scheduler.unstable_UserBlockingPriority,
+          () => {
+            ReactNoop.render(
+              <>
+                <Component
+                  label="a"
+                  getSnapshot={newGetSnapshot}
+                  mutableSource={mutableSource}
+                  subscribe={defaultSubscribe}
+                />
+                <Component
+                  label="b"
+                  getSnapshot={newGetSnapshot}
+                  mutableSource={mutableSource}
+                  subscribe={defaultSubscribe}
+                />
+              </>,
+              () => Scheduler.unstable_yieldValue('Sync effect'),
+            );
+          },
+        );
+        expect(Scheduler).toFlushAndYieldThrough([
+          'a:new:two',
+          'b:new:two',
+          'Sync effect',
+        ]);
+      });
+    });
+
+    // TODO (useMutableSource) Re-enable this test once workLoopSync can handle
+    // the thrown error without getting stuck in a cycle.
+    xit('should throw and restart render if source and snapshot are unavailable during a sync update', () => {
```

Comment:
This should work now if you rebase on master

Reply:
Just rebased and still fails. :(